### PR TITLE
name changes + removing references to activity when destroyed

### PR DESCRIPTION
**Pull request by berryve**

### CODE REVIEW

File: app/src/main/java/org/kinecosystem/kinit/view/transfer/TransferActivity.kt [hunk 1]
```diff
@@ -20,41 +19,34 @@ class TransferActivity : BaseActivity(), TransferActions {
...
     override fun onCreate(savedInstanceState: Bundle?) {
-        Log.d("###", "### on create TransferActivity")
-        Log.d("###", "### on create TransferActivity savedInstanceState " + savedInstanceState)
-
         KinitApplication.coreComponent.inject(this)
         super.onCreate(savedInstanceState)
         navigator = Navigator(this)
         setContentView(R.layout.single_fragment_layout)
-        app = intent.getParcelableExtra<EcoApplication>(APP_PARAM)
+        app = intent.getParcelableExtra(APP_PARAM)
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/view/transfer/TransferActivity.kt]: why twice?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/EcoAppsCategoryViewModel.kt [hunk 2]
```diff
@@ -23,18 +23,18 @@ class EcoAppsCategoryViewModel(private val context: Context, private val navigat
...
 
 
-    fun onItemClicked(app: EcoApplication) {
+    fun onItemClicked(app: EcosystemApp) {
         navigator.navigateTo(app)
     }
 
-    fun onActionBtnClicked(app: EcoApplication) {
+    fun onActionBtnClicked(app: EcosystemApp, context: Context) {
         if (app.isKinTransferSupported() && GeneralUtils.isAppInstalled(context, app.identifier)) {
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/EcoAppsCategoryViewModel.kt]: why did you pass the context?

File: app/src/main/java/org/kinecosystem/kinit/view/transfer/SendAmountFragment.kt [hunk 3]
```diff
@@ -53,11 +54,21 @@ class SendAmountFragment : BaseFragment() {
 
     }
 
+    override fun onDetach() {
+        super.onDetach()
+        model.onDetach()
+        transferActions = null
+    }
+
+    override fun onAttach(context: Context?) {
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/view/transfer/SendAmountFragment.kt]: ?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt [hunk 4]
```diff
@@ -4,21 +4,19 @@ import android.content.Context
...
+import org.kinecosystem.kinit.model.spend.EcosystemApp
 import org.kinecosystem.kinit.model.spend.getHeaderImageCount
 import org.kinecosystem.kinit.model.spend.getHeaderImageUrl
 import org.kinecosystem.kinit.model.spend.isKinTransferSupported
 import org.kinecosystem.kinit.navigation.Navigator
 import org.kinecosystem.kinit.util.GeneralUtils
 import javax.inject.Inject
 
-class AppViewModel(private val navigator: Navigator, private val app: EcoApplication, context:Context) : ViewPager.OnPageChangeListener {
+class AppViewModel(private val navigator: Navigator, private val app: EcosystemApp, context:Context) : ViewPager.OnPageChangeListener {
```
[the remark this answers is not in the document]
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt]: you need it in the constructor but don't hold it as a member var

File: app/src/main/java/org/kinecosystem/kinit/view/transfer/TransferActivity.kt (hunk 1 above)
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/view/transfer/TransferActivity.kt]: fixed

File: app/src/main/java/org/kinecosystem/kinit/view/transfer/SendAmountFragment.kt (hunk 3 above)
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/view/transfer/SendAmountFragment.kt]: fixed

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt]: removed the context

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferToAppViewModel.kt [hunk 5]
```diff
@@ -60,4 +60,12 @@ class TransferToAppViewModel(private val navigator: Navigator, val app: EcoAppli
     fun onSend(view: View?) {
         transferActions?.onStartTransferring(amount.get())
     }
+
+    fun onDetach() {
+        transferActions = null
+    }
+
+    fun onAttach(t: TransferActions) {
```
[the remark this answers is not in the document]
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferToAppViewModel.kt]: conflict with the member variable

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt [hunk 6]
```diff
@@ -4,21 +4,18 @@ import android.content.Context
...
+import org.kinecosystem.kinit.model.spend.EcosystemApp
 import org.kinecosystem.kinit.model.spend.getHeaderImageCount
 import org.kinecosystem.kinit.model.spend.getHeaderImageUrl
 import org.kinecosystem.kinit.model.spend.isKinTransferSupported
 import org.kinecosystem.kinit.navigation.Navigator
-import org.kinecosystem.kinit.util.GeneralUtils
 import javax.inject.Inject
 
-class AppViewModel(private val navigator: Navigator, private val app: EcoApplication, context:Context) : ViewPager.OnPageChangeListener {
+class AppViewModel(private val navigator: Navigator, private val app: EcosystemApp, isAppInstalled: Boolean) : ViewPager.OnPageChangeListener {
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt]: not sure it should be as param
its appview model role to do the check 
not the view that use it

**Pull request merged**